internal/lightcone/destruction/somethingirreplaceable/somethingirreplaceable.go

```diff
@@ -0,0 +1,91 @@
+package somethingirreplaceable
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Check key.Modifier = "something-irreplaceable"
+	Buff  key.Modifier = "kinship"
+)
+
+type state struct {
+	Heal     float64
+	DmgBonus float64
+}
+
+//Increases the wearer's ATK by 24%.
+//When the wearer defeats an enemy or is hit, immediately restores HP equal to 8% of the wearer's ATK.
+//At the same time, the wearer's DMG is increased by 24% until the end of their next turn.
+//This effect cannot stack and can only trigger 1 time per turn.
+
+func init() {
+	lightcone.Register(key.SomethingIrreplaceable, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        5,
+		Path:          model.Path_DESTRUCTION,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(Check, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAfterBeingAttacked: onAfterBeingAttacked,
+			OnTriggerDeath:       onTriggerDeath,
+		},
+	})
+
+	modifier.Register(Buff, modifier.Config{
+		StatusType: model.StatusType_STATUS_BUFF,
+	})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	atkBuff := 0.2 + 0.04*float64(lc.Imposition)
+
+	engine.AddModifier(owner, info.Modifier{
+		Name:   Check,
+		Source: owner,
+		Stats:  info.PropMap{prop.ATKPercent: atkBuff},
+		State: state{
+			Heal:     0.07 + 0.01*float64(lc.Imposition),
+			DmgBonus: 0.2 + 0.04*float64(lc.Imposition),
+		},
+	})
+}
+
+func onTriggerDeath(mod *modifier.ModifierInstance, target key.TargetID) {
+	conditions(mod)
+}
+
+func onAfterBeingAttacked(mod *modifier.ModifierInstance, e event.AttackEndEvent) {
+	conditions(mod)
+}
+
+func conditions(mod *modifier.ModifierInstance) {
+	heal := mod.State().(*state).Heal
+	dmgBonus := mod.State().(*state).DmgBonus
+
+	if mod.Engine().HasModifier(mod.Owner(), Buff) {
+
+	} else {
```

Comment:
Either: change to `if !mod.Engine().HasModifier(mod.Owner(), Buff)` instead of having blank if & else or do this to reduce depth
```go
if mod.Engine().HasModifier(mod.Owner(), Buff) {
  return
}
```